ledger/simulation/tracer.go

```diff
@@ -140,6 +140,12 @@ func (tracer *evalTracer) BeforeTxnGroup(ep *logic.EvalParams) {
 		tracer.result.TxnGroups[0].AppBudgetAdded = uint64(*ep.PooledApplicationBudget)
 	}
 
+	// Override transaction group budget if specified in request, retreve from tracer.result
+	if tracer.result.EvalOverrides.ExtraBudget != nil {
+		tracer.result.TxnGroups[0].AppBudgetAdded += *tracer.result.EvalOverrides.ExtraBudget
+		*ep.PooledApplicationBudget += int(*tracer.result.EvalOverrides.ExtraBudget)
```

Comment:
`ep.PooledApplicationBudget` will be nil before budget pooling, so I think it'd be best to guard this with an if statement. I think it's fine if we do nothing if budget pooling isn't enabled.

Reply:
wait... are you talking about budget pooling as a consensus param, here:
```go
	v30.EnableAppCostPooling = true
```

I am fine with adding an additional logic condition to guard the branch, but let me know if I am looking at the right thing.

Reply:
see [guard around ep.PooledApplicationBudget for nil ptr](https://github.com/algorand/go-algorand/pull/5354/commits/32abe6b682dc2fdb5f3bdcd454f0c5a480528e69)

Reply:
Yes that's right, thanks